src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -60,6 +60,18 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             MsalAccessTokenCacheItem cachedAccessTokenItem = await GetCachedAccessTokenAsync()
                 .ConfigureAwait(false);
 
+            // If using IMDSv2 (mTLS) and the mTLS cert is near expiry, skip the AT cache
+            if (cachedAccessTokenItem != null && ManagedIdentityClient.s_sourceName == ManagedIdentitySource.ImdsV2)
+            {
+                string identityKey = ServiceBundle.Config.ClientId;
```

Comment:
Naming is confusing. avoid using `identity` as it has too many meanings. How about `mtlsCacheKey` ?